compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1307,15 +1310,14 @@ pub fn catch_with_exit_code(f: impl FnOnce() -> interface::Result<()>) -> i32 {
 static ICE_PATH: OnceLock<Option<PathBuf>> = OnceLock::new();
 
 fn ice_path() -> &'static Option<PathBuf> {
+    ice_path_with_config(None)
```

Comment:
Doesn't this result in the behavior that calling `ice_path()` *then* calling `ice_path_with_config(Some(/*...*/))` will return the ICE path w/ no config? 

Could you leave some sort of comment saying that it's intended that `ice_path_with_config` has been called first, but this will still return a path in case an ICE is emitted before the session has been initialized?

Reply:
Alternatively, perhaps it's worthwhile to just parse the relevant env arg directly in `ice_path`?

Reply:
The relevant configuration is a compiler flag, not an environment variable, unfortunately :(

And yeah, if you call the `ice_path()` function first it would mess up the `_with_config(Some(..))` version. I wrote this with the assumption that `ice_path()` would only ever be called from the ice hook, so there's no chance of a subsequent call to `_with_config`. 

I can definitely add comments clarifying this.

Reply:
comment added, resolve if you're happy with it